apps/io-services-cms-webapp/src/publicator/__test__/request-publication-handler.test.ts

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable no-console */
+import { Context } from "@azure/functions";
+import { ServiceLifecycle, ServicePublication } from "@io-services-cms/models";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { Json } from "io-ts-types";
+import { describe, expect, it, vi, afterEach } from "vitest";
+import * as TE from "fp-ts/lib/TaskEither";
+import { handleQueueItem } from "../request-publication-handler";
+
+const createContext = () =>
+  ({
+    bindings: {},
+    executionContext: { functionName: "funcname" },
+    log: { ...console, verbose: console.log },
+  } as unknown as Context);
+
+const atimestamp = 1685529694747;
+const aService = {
+  id: "aServiceId",
+  data: {
+    name: "aServiceName" as NonEmptyString,
+    description: "aServiceDescription",
+    authorized_recipients: [],
+    max_allowed_payment_amount: 123,
+    metadata: {
+      address: "via tal dei tali 123",
+      email: "service@email.it",
+      pec: "service@pec.it",
+      scope: "LOCAL",
+    },
+    organization: {
+      name: "anOrganizationName",
+      fiscal_code: "12345678901",
+    },
+    require_secure_channel: false,
+  },
+} as unknown as ServiceLifecycle.definitions.Service;
+
+const anInvalidQueueItem = { mock: "aMock" } as unknown as Json;
+
+describe("Service Historicization Handler", () => {
+  afterEach(() => {
+    vi.restoreAllMocks();
+  });
+  it("handleQueueItem should return an Error if queueItem is invalid", async () => {
+    const context = createContext();
+
+    const mockFsmLifecycleClient =
+      {} as unknown as ServicePublication.FsmClient;
+
+    try {
+      await handleQueueItem(
+        context,
+        anInvalidQueueItem,
+        mockFsmLifecycleClient
+      )();
+    } catch (error) {
+      expect(error).toHaveProperty("message");
```

Comment:
this test will pass even if the exception is not throw. you need one between:

A) declare the number of expectations using [expect.assertions](https://jestjs.io/docs/expect#expectassertionsnumber)

B) call fail() after `handleQueueItem`

Reply:
Thanks for the suggestion, the test has been modified to fail in case the decoding of the queued message does not fail using the following vitest components:

- [expect.rejects](https://vitest.dev/api/expect.html#rejects)
- [toThrowError](https://vitest.dev/api/expect.html#tothrowerror)

```typescript
await expect(() =>
      handleQueueItem(context, anInvalidQueueItem, mockFsmLifecycleClient)()
    ).rejects.toThrowError("Error while parsing incoming message");
```